spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1594,6 +1598,34 @@ public void onMessage(Message message) {
 		}
 	}
 
+	private QueueingConsumer createQueueingConsumer(final String queueName, Channel channel) throws Exception {
+		channel.basicQos(1);
+		final CountDownLatch latch = new CountDownLatch(1);
+		QueueingConsumer consumer = new QueueingConsumer(channel) {
+
+			@Override
+			public void handleCancel(String consumerTag) throws IOException {
+				super.handleCancel(consumerTag);
+				latch.countDown();
```

Comment:
Looking to the `QueueingConsumer.handle(Delivery delivery)` logic, I'm not sure that we need this implementation. 
I may understand why we wait for the `handleConsumeOk`, but this `cancel` isn't clear to me.
It just don't make sense, because `nextDelivery()` is throwing exception in case of `cancel` anyway.
WDYT?